src/rgw/rgw_quota.cc

```diff
@@ -973,7 +973,11 @@ class RGWQuotaHandlerImpl : public RGWQuotaHandler {
       need_resharding = true;
       if (suggested_num_shards) {
         uint32_t obj_multiplier = 2;
-        if (is_multisite) {
+        // the 'is_multisite' will be false when the zone does not log data for replication
+        // such as single zone and archive zone types. But we do want archive zone
+        // to be able to scale up the shard count faster because bucket index can
+        // get considerably large as we preserve multiple versions of objects
+        if (is_multisite || driver->get_zone()->get_tier_type() == "archive") {
```

Comment:
i guess i don't follow the logic here. is there anything specific about archive zone that shouldn't apply to versioned buckets in general? doesn't `num_objs` already account for versions?

we scaled this up for multisite because the reshard events are expensive on the syncing side. what exactly are we trying to optimize for archive zone?

Reply:
> we still want archive zone to behave the same way as other zones in multisite would w.r.t resharding

i'm trying to understand why, though. we raised the multiplier for multisite zones because each reshard event adds overhead to other zones syncing from it. so we reshard bigger to reduce the frequency of reshard events

why is this necessary for archive zone, when nobody is syncing from it? is there a workload where this becomes a problem?

Reply:
oh, there were cases when we started seeing large omaps on archive zone alone with large data sets.. I think the 4x multiplier would help there.

Reply:
does resharding eventually catch up and resolve the 'large omap' issue?

Reply:
I think a manual reshard was run to resolve the issue after that.